connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala

```diff
@@ -160,13 +160,12 @@ object KustoDataSourceUtils {
     klog.debug(s"$reporter: $message")
   }
 
-  private[kusto] def extractSchemaFromResultTable(result: Iterable[JSONObject]): String = {
+  private[kusto] def extractSchemaFromResultTable(result: ArrayNode): String = {
 
     val tableSchemaBuilder = new StringJoiner(",")
-
-    for (row <- result) {
+    for (i <- 0 until result.size()) {
       // Each row contains {Name, CslType, Type}, converted to (Name:CslType) pairs
-      tableSchemaBuilder.add(s"['${row.getString("Name")}']:${row.getString("CslType")}")
+      tableSchemaBuilder.add(s"['${result.get(i).get("Name").asText()}']:${result.get(i).get("CslType").asText()}")
```

Comment:
These constants Name and CSLType are used in few places. Would it be okay to refactor this into a SchemaConstants file and refer it from there?